metecho/api/jobs.py

```diff
@@ -146,6 +154,106 @@ def _create_branches_on_github(
     return task_branch_name
 
 
+def create_repository(
+    project: Project,
+    *,
+    user: User,
+    dependencies: Iterable[str],
+    template_repo_owner: str = None,
+    template_repo_name: str = None,
+):
+    """
+    Given a local Metecho Project create and bootstrap the corresponding GitHub repository.
+    """
+    project.refresh_from_db()
+
+    try:
+        # Ensure the user is part of the org that owns the project
+        gh_user = gh_given_user(user)
+        user_orgs = [org.login for org in gh_user.organizations()]
+        if project.repo_owner not in user_orgs:
+            raise ValueError(
+                _(
+                    "Either you are not a member of the %(name)s organization "
+                    "or it hasn't installed the Metecho GitHub app"
+                )
+                % {"name": project.repo_owner}
+            )
+
+        # Create team & repository on GitHub
+        org = get_org_for_repo_creation(project.repo_owner)
+        team = org.create_team(f"{project} Team")
+        team.add_or_update_membership(user.username, role="maintainer")
+        for collaborator in project.github_users:
+            team.add_or_update_membership(collaborator["login"], role="member")
```

Comment:
I just realized we're doing this for the requesting user, but they may also select themselves in the list. We might want to remove them from the list, and/or add a note explaining that they'll automatically be added? Or if not, we should make sure this doesn't fail if they select themselves and cause `add_or_update_membership` to be run twice for the creating user.

Reply:
Question for SFDO: Do we want to maintain the maintainer/member distinction? If we do (and I think we should) the creator should be removed from the collaborator list in line 188. If not then we should get rid of line 186 and ensure the creator is always present in the list.

I've added this to the agenda

Reply:
> I'm using the full-access gh app from Keybase, but maybe it has something to do with the permissions of that app?

Determined in Slack this was due to the GH App not having access to all repos. Need to discuss with SFDO

Reply:
When I created a [new repo](https://github.com/orgs/jerivastest/teams/metecho-test-org-team/members?query=role%3Amaintainer) and selected you @jerivas, it added you as a maintainer (not member) -- is that because you own the org? Or an issue we should look into? (Can wait until after out discussion with SFDO.)

Reply:
Very weird! I ended up creating a few teams of my own and you always showed up as a `maintainer`, not `member`. Even when making direct GH API requests to update your team membership to `member` the response always came back with `maintainer`. I then went to the org settings and changed your role to Member instead of Owner and that made you appear as Member in all teams (even ones created before the change).

So I guess by virtue of being an org owner you must always be a team maintainer 🤷

Reply:
Ok -- something to note for them in our mtg tomorrow? I confirmed with the OddBird org that it works as expected if the selected user is an Org Member and not Owner.

Reply:
Added to the checklist

Reply:
SFDO confirmed we want the creator to be the maintainer and everyone else to be regular members. They also said it's okay if org owners are always maintainers.